pilot/pkg/networking/core/v1alpha3/listener_test.go

```diff
@@ -419,8 +422,8 @@ func verifyOutboundTCPListenerHostname(t *testing.T, l *xdsapi.Listener, hostnam
 
 func verifyInboundHTTPListenerServerName(t *testing.T, l *xdsapi.Listener) {
 	t.Helper()
-	if len(l.FilterChains) != 1 {
-		t.Fatalf("expected %d filter chains, found %d", 1, len(l.FilterChains))
+	if len(l.FilterChains) != 2 {
```

Comment:
It isn't clear to me why this changed. Why do we have two filter chains now? I thought we have the same filter chain, but use a distinct connectionManager so when we mutate one it doesn't mutate the others. Maybe I misunderstood the change though

Reply:
You are right. We still have the same filter chain and have separate connectionManager. 

Two filter chains are only used for test. I configured the `OnInboundFilterChains` to have two filter chains.